llama_stack/apis/prompts/prompts.py

```diff
@@ -0,0 +1,173 @@
+    prompt: str | None = Field(default=None, description="The system prompt with variable placeholders")
+    version: str = Field(description="Version string (integer start at 1 cast as string)")
+    prompt_id: str = Field(description="Unique identifier in format 'pmpt_<48-digit-hash>'")
+    variables: dict[str, str] | None = Field(
+        default_factory=dict, description="Variables for dynamic injection using {{variable}} syntax"
+    )
+
+    @field_validator("prompt_id")
+    @classmethod
+    def validate_prompt_id(cls, prompt_id: str) -> str:
+        if not isinstance(prompt_id, str):
+            raise TypeError("prompt_id must be a string in format 'pmpt_<48-digit-hash>'")
+
+        if not prompt_id.startswith("pmpt_"):
+            raise ValueError("prompt_id must start with 'pmpt_' prefix")
+
+        hex_part = prompt_id[5:]
+        if len(hex_part) != 48:
+            raise ValueError("prompt_id must be in format 'pmpt_<48-digit-hash>' (48 lowercase hex chars)")
+
+        for char in hex_part:
+            if char not in "0123456789abcdef":
+                raise ValueError("prompt_id hex part must contain only lowercase hex characters [0-9a-f]")
+
+        return prompt_id
+
+    @field_validator("version")
+    @classmethod
+    def validate_version(cls, prompt_version: str) -> str:
+        try:
+            int_version = int(prompt_version)
+            if int_version < 1:
+                raise ValueError("version must be >= 1")
+        except ValueError as e:
+            if "invalid literal" in str(e):
+                raise ValueError("version must be a string representation of an integer") from e
+            raise
+        return prompt_version
+
+    @classmethod
+    def generate_prompt_id(cls) -> str:
+        # Generate 48 hex characters (24 bytes)
+        random_bytes = secrets.token_bytes(24)
+        hex_string = random_bytes.hex()
+        return f"pmpt_{hex_string}"
+
+
+class CreatePromptRequest(BaseModel):
+    """Request model to create a prompt."""
+
+    prompt: str = Field(description="The prompt text content")
+    variables: dict[str, str] = Field(default_factory=dict, description="Variables for dynamic injection")
+
+
+class UpdatePromptRequest(BaseModel):
+    """Request model for updating a prompt."""
+
+    prompt: str = Field(description="The prompt text content")
+    variables: dict[str, str] = Field(default_factory=dict, description="Variables for dynamic injection")
+
+
+class ListPromptsResponse(BaseModel):
+    """Response model to list prompts."""
+
+    data: list[Prompt]
+
+
+@runtime_checkable
+@trace_protocol
+class Prompts(Protocol):
+    """Protocol for prompt management operations."""
+
+    @webmethod(route="/prompts", method="GET")
+    async def list_prompts(self) -> ListPromptsResponse:
+        """List all prompts.
+
+        :returns: A ListPromptsResponse containing all prompts.
+        """
+        ...
+
+    @webmethod(route="/prompts/{prompt_id:path}", method="GET")
```

Comment:
should we have an endpoint or option to list out all available versions of a prompt?